lab/requirements.md

```diff
@@ -0,0 +1,53 @@
+# OPI Lab Requirements
+
+## Access
+
+- Lab staff can access to all machines.
+- OPI members should access only to their own machines.
```

Comment:
I think OPI members can request access to any machines and reserve them. If we want to explicitly call out that we do not want `Vendor A` requesting access to `Vendor B` xPU devices, we should do that. I can see some people being wary of that, but if we don't have the level of trust required here, then I think we're in much deeper trouble.

Reply:
I would assume whatever devices are put in the OPI lab would be of the "general availability" variety.   If a vendor is not ready to release a device, or on the verge of doing so, maybe that device should be kept in that vendor's private lab.  CI can still run to access such devices.  Maybe put in a requirement to capture this.  "Server in OPI lab can communicate with a device in a vendor's lab through a secure access API."  Definition of this secure access API can be taken up as a separate topic, accounting for whatever security constraints folks may have.

Reply:
I think the key point is that no-one will ever send anything that isn't broad market to a shared lab. So all devices in the lab are by definition available to be purchased online by anyone.

Reply:
This goes back to the goal specified at the beginning of the file i.e. equipment is to be used for CI and functional validation. The OPI lab not intendent for exploratory purposes or for competitive testing. Public labs such as the Nvidia-launchpad have been created for that purpose and are heavy handed in security. OPI should not take that role on.

We will have to call out that only vendors can access their equipment for troubleshooting and tuning. Assuming implicit trust is not possible and will likely not be approved by management.

Reply:
If the only point of this lab is to run a set of tests on vendor HW while not allowing anyone but that vendor to login to the vendor specific HW, why are we doing this? I would recommend we just have each vendor run their own tests in their own labs and report back to the CI system on each PR. That is a solved problem that has been done for 10+ years now, and makes this much easier.

Reply:
Vendor testing is an acceptable solution. It has been rather difficult to justify the need for another public lab when that process is already being handled (thoroughly vetted for security purposes by management) for the NVIDIA launchpad. We can certainly publish a recipe for validating OPI specs in the NVIDIA launchpad.

The type of restricted use defined here was considered an acceptable middle-ground as we already have a precedent with the DPDK community successfully using it.

Reply:
It's unclear to me why we would use the OPI funding to run tests on a bunch of cards which only each vendor can access. A better solution is to just have each vendor do this themselves.

Reply:
what about INTEROP ?

Reply:
I agree with Kyle here.  The most important point of having a public lab was to allow developers to create multi-vendor capable applications (Infrastructure Services) by allowing them access to instances of each vendors products for portability and compatibility reasons.  To enable a broad xPU ecosystem, with a rich developer community, it seems unwise, and onerous, to assume each developer must create his own multi-vendor lab (and may not be able to anyway do to lack of product availability or his own resource limitations).  If each vendor chooses to establish a public lab of only their products (like Nvidia has), that also addresses the need, but requires a developer to navigate each lab as a separate effort adding friction and slowing development.  If we can get two or more vendors to support a common lab, than it is interesting.  If we have one, or none, than their is not much point.  I think an OPI lab, non-public, for CI testing only, done by OPI, is a different question and we should not conflate these two conversations (although CI testing could be done in the Public lab).

Reply:
@timmichels That is fair. What you are clearly asking for is an exploratory lab that "any" developer can use. IMO any developer can become an OPI community developer and OPI TSC/BOD doesn't get to vet each one of them; that is in direct conflict with "community" plus I imagine no one at OPI has the cycles to that.

As you said that sort of "multi-vendor exploratory development lab" (isn't that the dream of every developer!) is very different from a CI-lab. We should break the discussion up into -
1. CI lab
2. Multi-vendor development lab - anybody can access to develop and test. Getting such a lab approved is certainly beyond my powers and best addressed via the TSC. I will suggest that standing up such a lab is less likely and not all vendors are going to be willing to participate.

Interop can theoretically be addressed in a CI-lab.

Reply:
@AnuradhaKaruppiah I don't see the point in making OPI pay to have someone run a lab where we lock-down each vendor to their own equipment. It makes no sense. Each vendor can do that themselves. OPI can produce the tests they run and report back on pull requests, for example. Also, how would we restrict access? If I'm part of the CI team, for example, it would mean I would not have access to AMD, Marvell, or NVIDIA cards. And vice-versa. This is a serious short coming and devalues any attempt at a CI lab, run for and paid by OPI, with very limited access to the actual hardware.

The second option is the more interesting option, and I agree with Tim's comments and I was also under the impression this is the direction we were heading.

Further, I don't understand how the vendors could stop OPI from just procuring cards through some third party and installing them and setting them up. Are we saying we will police this process and prevent this from happening?

Reply:
@mestery Yes, all valid questions. As this thread doesn't seem to converging maybe you can move this to TSC and/or BOD; primarily to identify the goals of the OPI lab? Rest of the content in the doc can be reviewed/updated iteratively later. @tmonjalo and I will bring this to the notice of the NVIDIA reps as well.

Reply:
Yes, the main original reason we wanted a lab was not for the hardware vendors to use, it was for software vendors.  Presumably each hardware vendor has their own lab to use for testing/certification.  The point of the community lab was so that projects/companies could have access to OPI compatible hardware without needing to build their own multi-vendor lab.

Reply:
The goal of the lab was not written, that's why I am starting the file goals.md
I will split the PR in 2, so we can merge a first part while keeping a second part for debate.

Reply:
@tmonjalo so this PR was split in two. I see the automated tests section of the PR is here. Where is the section that discussed the goal of using the lab beyond the automated tests and using it for exploratory purposes? Is that to be a future PR? Or is the split you undertook to split the goals and the requirements from one another? If the former I will add more comments about the exploratory purpose goals to that future PR. If the latter I will add comments to the goals.md

Reply:
I want to have a consensus baseline first.
I'm collecting comments and will update this PR to its minimal form reaching consensus.
So we can propose new PR on top once this one is merged.
Let's close this one quickly so you can propose a PR about exploratory goals.
Are you OK with the steps?